ws-client/src/client.rs

```diff
@@ -121,10 +121,32 @@ impl WsClient {
 			.map_err(|e| Error::TransportError(Box::new(e)))?;
 
 		async_std::task::spawn(async move {
-			background_task(sender, receiver, from_front, max_capacity_per_subscription, max_concurrent_requests).await;
+			let result =
+				background_task(sender, receiver, from_front, max_capacity_per_subscription, max_concurrent_requests)
+					.await;
+			panic!("terminating client: {:?}", result)
 		});
 		Ok(Self { to_back, request_timeout })
 	}
+
+	/// Initializes a new WebSocket client and returns the background task
+	///
+	/// Fails when the URL is invalid.
+	pub async fn new_and_background(config: WsConfig<'_>) -> Result<(WsClient, impl futures::Future), Error> {
```

Comment:
I'm not super excited about this API seems overly complicated to use i.e, both create a client and manage the background task, maybe I'm biased because I prefer my approach in #234 (should be quite easy to extend and add a restart method I think)

Then if you for example call a request and the background is terminate you still get `Error::Internal` which is not good. 

//cc @maciejhirsz @dvdplm thoughts?

Reply:
Your approach probably makes more sense, this was just something that worked for me when testing locally. That being said there is probably some advantage to a configurable executor.

Reply:
Yeah, configurable executor would be neat indeed but not really a priority and everything doesn't work across different runtimes (however the ws client should work).

Do you mind filing an issue for it? :)